From fe07b3b1258e8710fee95f61daf082ca36686422 Mon Sep 17 00:00:00 2001
From: Robert Richter <rrichter@marvell.com>
Date: Wed, 29 May 2019 14:24:29 +0000
Subject: [PATCH] kabi: arm64: cpuhotplug: Reuse other arch's cpuhp_state

Patch-mainline: Never, kabi fix 
References: fate#323052,bsc#1117114

Do not introduce new enum values as this will cause a kernel ABI
breakage. Instead reuse other arch's cpuhp_state.

Signed-off-by: Robert Richter <rrichter@marvell.com>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 include/linux/cpuhotplug.h | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/linux/cpuhotplug.h b/include/linux/cpuhotplug.h
index edb97952b..9b20b5828 100644
--- a/include/linux/cpuhotplug.h
+++ b/include/linux/cpuhotplug.h
@@ -145,7 +145,7 @@ enum cpuhp_state {
 	CPUHP_AP_PERF_ARM_L2X0_ONLINE,
 	CPUHP_AP_PERF_ARM_QCOM_L2_ONLINE,
 	CPUHP_AP_PERF_ARM_QCOM_L3_ONLINE,
-	CPUHP_AP_PERF_ARM_CAVIUM_TX2_UNCORE_ONLINE,
+	/* kernel ABI: CPUHP_AP_PERF_ARM_CAVIUM_TX2_UNCORE_ONLINE, */
 	CPUHP_AP_PERF_POWERPC_NEST_IMC_ONLINE,
 	CPUHP_AP_PERF_POWERPC_CORE_IMC_ONLINE,
 	CPUHP_AP_PERF_POWERPC_THREAD_IMC_ONLINE,
@@ -159,6 +159,12 @@ enum cpuhp_state {
 	CPUHP_ONLINE,
 };
 
+#ifdef CONFIG_ARM64
+/* Reuse other arch's entries to avoid kernel ABI breakage */
+#define CPUHP_AP_PERF_ARM_CAVIUM_TX2_UNCORE_ONLINE	\
+		CPUHP_AP_PERF_POWERPC_NEST_IMC_ONLINE
+#endif
+
 int __cpuhp_setup_state(enum cpuhp_state state,	const char *name, bool invoke,
 			int (*startup)(unsigned int cpu),
 			int (*teardown)(unsigned int cpu), bool multi_instance);
-- 
2.11.0

